beacon_node/lighthouse_network/src/discovery/mod.rs

```diff
@@ -1036,12 +1038,107 @@ impl<TSpec: EthSpec> NetworkBehaviour for Discovery<TSpec> {
             FromSwarm::DialFailure(DialFailure { peer_id, error, .. }) => {
                 self.on_dial_failure(peer_id, error)
             }
+            FromSwarm::NewListenAddr(ev) => {
+                /* TODO(jmcph4): plumb user override status into config somehow */
+                let addr: &Multiaddr = ev.addr;
+                let listener_id: ListenerId = ev.listener_id;
+
+                trace!(self.log, "Received NewListenAddr event from swarm"; "listener_id" => ?listener_id, "addr" => ?addr);
+
+                let ip: IpAddr = match addr.iter().next() {
+                    Some(Protocol::Ip4(ip)) => IpAddr::V4(ip),
+                    Some(Protocol::Ip6(ip)) => IpAddr::V6(ip),
+                    _ => {
+                        debug!(self.log, "Encountered unacceptable multiaddr for listening (no IP)"; "addr" => ?addr);
+                        return;
+                    }
+                };
+
+                let update_enr = |ip: IpAddr, transport: Protocol| {
+                    /* this closure shouldn't exist but `Ipv4Addr::is_global` is Nightly at the moment */
+                    let is_global = |ip: Ipv4Addr| {
+                        !(ip.is_broadcast()
+                            || ip.is_documentation()
+                            || ip.is_link_local()
+                            || ip.is_loopback()
+                            || ip.is_multicast()
+                            || ip.is_private()
+                            || ip.is_unspecified())
+                    };
+
+                    let insert_into_enr = |key: &str, port: &u16| {
+                        if let Err(e) = self.discv5.enr_insert(key, port) {
+                            warn!(self.log, "Failed to write to ENR"; "error" => ?e);
+                        }
+                    };
+
+                    let mut should_update_ip: bool = true;
```

Comment:
I think we should never update the ENR IP here. 

A few reasons:
1. The local address we are binding to (i.e the address that libp2p will return) will almost never be a globally routable address. Most computers will be behind a NAT. Unless they are using ipv6 etc. But in any case its better to let discovery handle the IP field of the ENR. 
2. Discovery sets this field based on how others connect to us. We should not interfere with this I dont think.

We should only really set the TCP or QUIC fields of the ENR based on libp2p's response.